docs/architecture/adr-056-proving-amnesia-attack.md

```diff
@@ -0,0 +1,106 @@
+# ADR 056: Proving flip flopping attacks
+
+## Changelog
+
+- 02.04.20: Initial Draft
+- 06.04.20: Second Draft
+
+## Context
+
+Whilst most created evidence of malicious behaviour is self evident such that any individual can verify them independently there are types of evidence, known as global evidence, that require further collaboration from the network in order to accumulate enough information to create evidence that is individually verifiable and can therefore be processed through consensus. Fork Accountability as a whole constitutes of detection, proving and punishing. This ADR addresses how to prove infringement from global evidence that is sent to a full node.
+
+The currently only known form of global evidence stems from [flip flopping](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md#flip-flopping) attacks. The schematic below explains one scenario where an amnesia attack, a form of flip flopping, can occur such that C1 and C2 commit different blocks.
+
+![](../imgs/tm-amnesia-attack.png)
+
+This creates a fork on the main chain.  Back to the past, another form of flip flopping, creates a light fork (capable of fooling those not involved in consensus), in a similar way, with F taking the precommits from C1 and forging a commit from them.
```

Comment:
note: amnesia can be used to create a light fork too

Reply:
Isn't that what "back to the past" is?

Reply:
no, in back to the past, you're taking advantage of the fact that you haven't signed anything in this round earlier (i.e you have not seen 2/3+ votes, hence no amnesia)

Reply:
okay, and so how would an amnesia light fork attack be different other than that the malicious nodes didn't prevote on that round?